tools/mason/MasonUtils.chpl

```diff
@@ -518,6 +519,189 @@ proc getProjectType(): string {
   return tomlFile["brick"]!["type"]!.s;
 }
 
+proc getDepToml(depName: string, depVersion: string) throws {
+  const pattern = compile(depName, ignoreCase=true);
+
+  var packages: list(string);
+  var versions: list(string);
+  var registries: list(string);
+  var results: list(string);
+  for registry in MASON_CACHED_REGISTRY {
+    const searchDir = registry + "/Bricks/";
+
+    for dir in listdir(searchDir, files=false, dirs=true) {
+      const name = dir.replace("/", "");
+      if pattern.search(name) {
+        const ver = findLatest(searchDir + dir);
+        const versionZero = new VersionInfo(0, 0, 0);
+        if ver != versionZero {
+          results.append(name + " (" + ver.str() + ")");
+          packages.append(name);
+          versions.append(ver.str());
+          registries.append(registry);
+        }
+      }
+    }
+  }
+
+  if results.size > 0 {
+    const brickPath = '/'.join(registries[0], 'Bricks', packages[0], versions[0]) + '.toml';
+    const openFile = openreader(brickPath);
+    const toml = parseToml(openFile);
+
+    return toml;
+  } else {
+    throw new owned MasonError("No TOML file in registry for " + depName);
+  }
+}
+
+
+/* Search TOML files within a package directory to find the latest package
+   version number that is supported with current Chapel version */
+proc findLatest(packageDir: string): VersionInfo {
+  use Path;
+
+  var ret = new VersionInfo(0, 0, 0);
+  const suffix = ".toml";
+  const packageName = basename(packageDir);
+  for manifest in listdir(packageDir, files=true, dirs=false) {
+    // Check that it is a valid TOML file
+    if !manifest.endsWith(suffix) {
+      var warningStr = "File without '.toml' extension encountered - skipping ";
+      warningStr += packageName + " " + manifest;
+      stderr.writeln(warningStr);
+      continue;
+    }
+
+    // Skip packages that are out of version bounds
+    const chplVersion = getChapelVersionInfo();
+
+    const manifestReader = openreader(packageDir + '/' + manifest);
+    const manifestToml = parseToml(manifestReader);
+    const brick = manifestToml['brick'];
+    var (low, high) = parseChplVersion(brick);
+    if chplVersion < low || chplVersion > high then continue;
+
+    // Check that Chapel version is supported
+    const end = manifest.size - suffix.size;
+    const ver = new VersionInfo(manifest[0..<end]);
+    if ver > ret then ret = ver;
+  }
+  return ret;
+}
+
+
+proc parseChplVersion(brick: borrowed Toml?): (VersionInfo, VersionInfo) {
```

Comment:
As you are moving these functions, could you add a short comment about what they are used for? Maybe something like `/* Reads the chapel version specified by a mason project's toml file and returns the min and max compatible versions */`